integrations/sklearn/neptune_sklearn/__init__.py

```diff
@@ -0,0 +1,1190 @@
+from scikitplot.metrics import plot_precision_recall
+from sklearn.base import is_regressor, is_classifier
+from sklearn.cluster import KMeans
+from sklearn.metrics import explained_variance_score, max_error, mean_absolute_error, r2_score, \
+    precision_recall_fscore_support
+from yellowbrick.classifier import ClassificationReport, ConfusionMatrix, ROCAUC, ClassPredictionError
+from yellowbrick.cluster import SilhouetteVisualizer, KElbowVisualizer
+from yellowbrick.model_selection import FeatureImportances
+from yellowbrick.regressor import ResidualsPlot, PredictionError, CooksDistance
+
+import neptune.new as neptune
+
+
+def create_regressor_summary(regressor, X_train, X_test, y_train, y_test, nrows=1000, log_charts=True):
+    """Create sklearn regressor summary.
+
+    This method creates regressor summary that includes:
+
+    * all regressor parameters,
+    * pickled estimator (model),
+    * test predictions,
+    * test scores,
+    * model performance visualizations.
+
+    Returned ``dict`` can be assigned to the experiment's namespace defined by the user (see example below).
+
+    Regressor should be fitted before calling this function.
+
+    Tip:
+        Check Sklearn-Neptune integration
+        `documentation <https://docs-beta.neptune.ai/essentials/integrations/machine-learning-frameworks/sklearn>`_
+        for the full example.
+
+    Args:
+        regressor (:obj:`regressor`):
+            | Fitted sklearn regressor object
+        X_train (:obj:`ndarray`):
+            | Training data matrix
+        X_test (:obj:`ndarray`):
+            | Testing data matrix
+        y_train (:obj:`ndarray`):
+            | The regression target for training
+        y_test (:obj:`ndarray`):
+            | The regression target for testing
+        nrows (`int`, optional, default is 1000):
+            | Log first ``nrows`` rows of test predictions.
+        log_charts (:bool:, optional, default is ``True``):
+            | If ``True``, calculate and log chart visualizations.
+            |
+            | NOTE: calculating visualizations is potentially expensive depending on input data and regressor, and
+            | may take some time to finish.
+            |
+            | This is equivalent to calling ``log_learning_curve_chart``, ``log_feature_importance_chart``,
+            | ``log_residuals_chart``, ``log_prediction_error_chart``, ``log_cooks_distance_chart``
+            | functions from this module.
+
+    Returns:
+        ``dict`` with all summary items.
+
+    Examples:
+        Log random forest regressor summary.
+
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            rfr = RandomForestRegressor()
+            rfr.fit(X_train, y_train)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['summary/random_forest'] = npt_utils.create_regressor_summary(rfr, X_train, X_test, y_train, y_test)
```

Comment:
Information architecture wise shouldn't it be exp['random_forest/summary']?

Reply:
corrected.